src/main/java/org/opentripplanner/middleware/models/ApiUser.java

```diff
@@ -87,4 +87,32 @@ public void createApiKey(String usagePlanId, boolean persist) throws CreateApiKe
     public static ApiUser userForApiKey(String apiKeyId) {
         return Persistence.apiUsers.getOneFiltered(Filters.elemMatch("apiKeys", Filters.eq("keyId", apiKeyId)));
     }
+
+    /**
+     * @return the first {@link ApiUser} found with an {@link ApiKey#value} in {@link #apiKeys} that matches the
+     * provided apiKeyValue.
+     */
+    public static ApiUser userForApiKeyValue(String apiKeyValue) {
+        return Persistence.apiUsers.getOneFiltered(Filters.elemMatch("apiKeys", Filters.eq("value", apiKeyValue)));
+    }
+
+    /**
+     * Shorthand method to determine if an API user has an API key.
+     */
+    public boolean hasKey(String apiKeyId) {
+        return apiKeys
+            .stream()
+            .anyMatch(apiKey -> apiKeyId.equals(apiKey.keyId));
+    }
+
+    /**
+     * Shorthand method to determine if an API user has an API token (value).
+     */
+    public boolean hasToken(String apiKeyValue) {
```

Comment:
We don't really call the API key a "token" anywhere else.

Reply:
Updated.